src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import counter from "./Counter.module.scss";
+import { CounterProps, CounterState } from "../../models";
+
+class Counter extends React.Component<CounterProps, CounterState> {
+  constructor(props: CounterProps) {
+    super(props);
+    this.state = {
+      value: props.initialValue,
+    };
+  }
+
+  increment = () => {
+    this.setState((state) => ({
+      value: state.value + 1,
+    }));
+  };
+
+  decrement = () => {
+    this.setState((state) => ({
+      value: state.value - 1,
+    }));
+  };
+
+  render() {
+    return React.createElement(
+      "div",
+      { className: counter.container },
+      React.createElement(
+        "button",
+        {
+          onClick: this.increment,
```

Comment:
I guess it should be the other way around: `this.decrement` for "-" and `this.increment` for "+".